library/components/TTooltip.vue

```diff
@@ -1,64 +1,110 @@
-            left: "top-2 -right-[7px]",
-            right: "top-2 -left-[7px]",
-        }
+		positions: {},
+		showTooltip: false,
+        isFocused: false,
 	}),
 	mounted() {
-		this.placeTooltip();
+		onClickOutside(this.$refs.tooltipContainer, () => {
+			this.hide();
+		});
 	},
 	methods: {
 		placeTooltip() {
-
-            // Place tooltip near parent according to `position` prop.
-			const parent = this.$parent.$el;
-            const tooltipPositions = {}
-			if (parent) {
-				const position = parent.getBoundingClientRect();
+			// Calculate and place both tooltip and arrow.
+			const triggerElement = this.$refs.trigger;
+			const tooltipElement = this.$refs.tooltip;
+			const tooltipPositions = {};
+			const positions = { arrow: {}, tooltip: {} };
+			if (triggerElement && tooltipElement) {
+				const position = triggerElement.getBoundingClientRect();
 				if (this.position === "top") {
-					// Push from bottom
-					tooltipPositions.top = "-" + (position.bottom + 15) + "px";
+					positions.tooltip.top = position.top - (tooltipElement.clientHeight + 10) + "px";
+					positions.tooltip.left = position.left - tooltipElement.clientWidth / 2 + "px";
+					positions.arrow.top = position.top - 20 + "px";
+					positions.arrow.left = position.left + "px";
 				} else if (this.position === "bottom") {
-					// Push from top
-					tooltipPositions.top = (parent.clientHeight + 15) + "px";
+					positions.tooltip.top = position.top + triggerElement.clientHeight + 15 + "px";
+					positions.tooltip.left = position.left - tooltipElement.clientWidth / 2 + "px";
+					positions.arrow.top = position.top + triggerElement.clientHeight + 10 + "px";
+					positions.arrow.left = position.left + "px";
 				} else if (this.position === "left") {
-                    // Place on left side
-                    tooltipPositions.right = (position.width + 15) + "px";
-                } else {
-                    // Place on right side
-                    tooltipPositions.left = (position.width + 15) + "px";
-                }
+					positions.tooltip.top = position.top - tooltipElement.clientHeight / 2 + "px";
+					positions.tooltip.left = position.left - tooltipElement.clientWidth - 15 + "px";
+					positions.arrow.top = position.top + "px";
+					positions.arrow.left = position.left - 25 + "px";
+				} else {
+					positions.tooltip.top = position.top - tooltipElement.clientHeight / 2 + "px";
+					positions.tooltip.left = position.left + triggerElement.clientWidth + 15 + "px";
+					positions.arrow.top = position.top + "px";
+					positions.arrow.left = position.left + triggerElement.clientWidth + 10 + "px";
+				}
 			}
-            this.tooltipPositions = tooltipPositions;
+			this.positions = positions;
+		},
+		handleMouseOver() {
+			if (this.isFocused) return;
+			this.show();
+		},
+		handleMouseLeave() {
+			if (this.isFocused) return;
+			this.hide();
+		},
+		handleClick() {
+            this.isFocused = true;
+            this.show();
+		},
```

Comment:
It might be hard to keep `isFocused` up to date with the focused state of an element.

How do you feel about changing tooltip to provide a prop (`open`), to force the tooltip open. If I recall rightly you had something like this in an earlier implementation. The implementation of ProductCL is a good example: https://github.com/PatchUI/ProductCL/blob/main/src/components/BaseTooltip/BaseTooltip.vue#L115

Tooltip can still be responsible for its mouseover/mouseleave and potentially a click. However, a trigger element can tell the tooltip to display when its native html events are triggered

for example, focusing a button:
```
<tooltip :open="showTooltip">
  <button @focus="showTooltip = true" @blur="showTooltip=false"></button>
</tooltip>
```

That way, you can rely on the `open` prop inside the tooltip to decide whether to `return` on the mouse events and click events, and leverage native DOM events to open and close the tooltip when you have them available to you.

Reply:
This makes sense but the problem is the page files like `issues_detail.html` do not support javascript at all, this might not work for them and will make this feature exclusive to library or visualization components.

Also the reason I removed it was also because it might require more syntax for non-tech users, like Kennan or Ezra might not like this haha!

Though I still want to keep this in mind, maybe we can merge them in future.